std/stdio.d

```diff
@@ -4222,6 +4258,39 @@ private size_t readlnImpl(FILE* fps, ref char[] buf, dchar terminator = '\n')
     return buf.length;
 }
 
+unittest
+{
+    auto deleteme = testFilename();
+    scope(exit) std.file.remove(deleteme);
+
+    std.file.write(deleteme, "abcd\n0123456789abcde\n1234\n");
+    File f = File(deleteme, "rb");
+
+    char[] ln = new char[2];
+    assert(ln.capacity > 5);
```

Comment:
> We shouldn't assert any behavior of the array cache or the GC implementation.

A failing test reveals a bad interaction between the code here and the GC code, so it cannot be added as a unittest without these details , e.g. the "should not reallocate" optimization depends on a minimal capacity. The alternative would be to always pass the test if the initial assumption about the capacity is not correct.